### PR TITLE
Added auto-loader for plugin logger, including plugin updates

### PR DESCRIPTION
**Pull request by TheBurchLog**
Created a generic monitoring tool for files. Then set this to monitor the logging-config.yaml and spawn an event when it changes. This event is captured by the Log.py handler and reloads the file. It also spawns commands to local plugins to reload their logging if they are Running. That way the changes can be immediate.

Easiest way to test this is to add and remove the file logging option. If you change the file directory or name you will see that change happen semi-immediately. 

The cool thing is now the logging file doesn't have to exist at startup. They can create it later and then those configurations will be applied automatically. 

Depends on: https://github.com/beer-garden/brewtils/pull/230
Fixes https://github.com/beer-garden/beer-garden/issues/532

### CODE REVIEW

**Comment by hazmat345** [comment]
@TheBurchLog it's cool that you're able to reload the configurations, but that shouldn't happen automatically. Reloading the logging configuration of running plugins needs to be initiated by an admin user.

**Comment by TheBurchLog** [comment]
@hazmat345 
ok.... I will remove the auto reload of configurations but I am going to leave everything stubbed out so a follow on PR can enable it through the UI.

**Comment by TheBurchLog** [comment]
@hazmat345 Going to merge this in.

File: src/app/beer_garden/log.py [hunk 1]
```diff
@@ -169,7 +171,7 @@ def load(cls, filename: str, default_config) -> None:
         Returns:
             None
         """
-        if filename:
+        if filename and os.path.exists(filename):
```
Comment (hazmat345) [inline src/app/beer_garden/log.py:174]: This needs to be backed out. If an incorrect file is specified an error should be raised. This change will make the system silently fall back to the default configuration.

Comment (TheBurchLog) [reply src/app/beer_garden/log.py:174]: What if instead we still raise the error but allow them to proceed with the defaults. That way if they for some reason go back later and add the configuration they don't need to restart Beer Garden to apply it.

Comment (TheBurchLog) [reply src/app/beer_garden/log.py:174]: I backed it out and created a new PR for that change. #587

**Pull request merged**